### PR TITLE
[Coverage] Fix the strategy and testcase generation for coverage

### PR DESCRIPTION
**Pull request by ChenfengWei0**

### CODE REVIEW

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
@ChenfengWei0: could you also please add some regression tests in C++?

**Comment by ChenfengWei0** [comment]
> @ChenfengWei0: could you also please add some regression tests in C++?

Done

**Review by lucasccordeiro — APPROVED** [review]

File: regression/goto-coverage/Testing/Temporary/LastTest.log
Comment (intrigus-lgtm) [inline regression/goto-coverage/Testing/Temporary/LastTest.log:1]: What is this needed for?

**Review by intrigus-lgtm — CHANGES_REQUESTED** [review]

File: regression/goto-coverage/Testing/Temporary/LastTest.log
Comment (ChenfengWei0) [reply regression/goto-coverage/Testing/Temporary/LastTest.log:1]: Local ctest log file. Forgot to delete…

**Comment by lucasccordeiro** [comment]
Thanks for submitting this PR, @ChenfengWei0.

**Pull request merged**